src/osp/scientific/kepler.cpp

```diff
@@ -0,0 +1,436 @@
+    }
+
+    double hyperbolic_eccentric_anomaly_to_true_anomaly(const double &ecc, const double &H)
+    {
+        return 2 * atan(sqrt((ecc + 1) / (ecc - 1)) * tanh(H / 2));
+    }
+
+    double true_anomaly_to_eccentric_anomaly(const double &ecc, const double &v)
+    {
+        return 2.0 * std::atan(std::sqrt((1 - ecc) / (1 + ecc)) * std::tan(v / 2.0));
+    }
+
+    double true_anomaly_to_hyperbolic_eccentric_anomaly(const double &ecc, const double &v)
+    {
+        return 2.0 * std::atanh(std::clamp(std::sqrt((ecc - 1) / (ecc + 1)) * std::tan(v / 2.0), -1.0, 1.0));
+    }
+
+    double mean_motion(double mu, double a)
+    {
+        return sqrt(mu / (a * a * a));
+    }
+
+    double get_orbiting_radius(double ecc, double semiMajorAxis, double trueAnomaly)
+    {
+        return semiMajorAxis * (1 - ecc * ecc) / (1 + ecc * std::cos(trueAnomaly));
+    }
+
+
+
+
+    double KeplerOrbit::get_true_anomaly_from_eccentric(double eccentric) const
+    {
+        if (is_elliptic())
+        {
+            return eccentric_anomaly_to_true_anomaly(m_eccentricity, eccentric);
+        }
+        else if (is_hyperbolic())
+        {
+            return hyperbolic_eccentric_anomaly_to_true_anomaly(m_eccentricity, eccentric);
+        }
+    }
+
+    double KeplerOrbit::get_eccentric_anomaly(double time) const
+    {
+        if (is_elliptic())
+        {
+#if 1 // Use standard kepler solver
+            double nu = mean_motion(m_gravitationalParameter, m_semiMajorAxis);
+            double Mt = m_meanAnomalyAtEpoch + (time - m_epoch) * nu;
+            Mt = wrap_angle(Mt);
+            return solve_kepler_elliptic(Mt, m_eccentricity, 20000);
+#else // Use universal kepler solver
+            double chi = solve_kepler_universal(m_r0, m_v0, m_semiMajorAxis, time - m_epoch, m_gravitationalParameter, 200);
+            double E = chi / std::sqrt(m_semiMajorAxis) + m_E0;
+            return wrap_angle(E);
+#endif
+        }
+        else if (is_hyperbolic())
+        {
+#if 1 // Use standard kepler solver
+            double nu = mean_motion(m_gravitationalParameter, -m_semiMajorAxis);
+            double Mt = m_meanAnomalyAtEpoch + (time - m_epoch) * nu;
+            // Mt = wrap_angle(Mt);
+            return solve_kepler_hyperbolic(Mt, m_eccentricity, 20000);
+#else // Use universal kepler solver
+            double chi = solve_kepler_universal(m_r0, m_v0, m_semiMajorAxis, time - m_epoch, m_gravitationalParameter, 2000);
+            double F = chi / std::sqrt(-m_semiMajorAxis) + m_F0;
+            return F;
+#endif
+        }
+    }
+
+    void KeplerOrbit::get_fixed_reference_frame(Vector3d &radial, Vector3d &transverse, Vector3d &outOfPlane) const
+    {
+        double a = m_argumentOfPeriapsis;
+        double b = m_longitudeOfAscendingNode;
+        double c = m_inclination;
+
+        radial = Vector3d(std::cos(a) * std::cos(b) - std::sin(a) * std::sin(b) * std::cos(c),
```

Comment:
I feel like it's practical to just have `using std::sin;` and `using std::cos;` at the top of the file, this feels a little noisy.